refinery/ui/source/js/user-file-browser/services/factory.js

```diff
@@ -79,26 +81,70 @@
       return defs;
     }
 
-    function createData (solrNodes) {
+    function normalizeName (attribute) {
+      return attribute.display_name.toLowerCase();
+    }
+
+    function createData (solrAttributes, solrNodes) {
+      var mapInternalToDisplay = {};
+      solrAttributes.forEach(function (attribute) {
+        mapInternalToDisplay[attribute.internal_name] =
+            normalizeName(attribute);
+      });
+
       var data = [];
-      for (var i = 0; i < solrNodes.length; i++) {
-        var node = solrNodes[i];
-        var url = node.REFINERY_NAME_6_3_s;
-        data.push({
-          url: url,
-          technology: 'TODO',
-          filename: url ? decodeURIComponent(url.replace(/.*\//, '')) : '',
-          organism: node.organism_Characteristics_6_3_s,
-          date: 'TODO',
-          antibody: node.antibody_Factor_Value_6_3_s,
-          cell_type: node.cell_line_Characteristics_6_3_s,
-          published: 'TODO',
-          accession: 'TODO',
-          genotype: 'TODO',
-          owner: 'TODO'
+      solrNodes.forEach(function (node) {
+        var row = {};
+        var internalNames = Object.keys(node);
+        internalNames.forEach(function (internalName) {
+          var display = mapInternalToDisplay[internalName];
+          row[display] = node[internalName];
         });
-      }
+        row.url = row.name;
```

Comment:
URLs are pulled from a separate API and not derived from names. It'd be great if the user/files api just provided them for you instead of performing a different request.